server/app/models/article.rb

```diff
@@ -0,0 +1,30 @@
+class Article < ApplicationRecord
+  # Relationships
+  belongs_to :user
+
+  # ActiveStorage relationship
+  has_one_attached :image
+
+
+  # Roles
+  enum status: [:pending, :approved, :rejected]
+
+  # Validations
+  validates :title, presence: true
+  validates :body, presence: true
+
+
+  # Helper methods
+  def image_url
+    Rails.application.routes.url_helpers.url_for(image) if image.attached?
+  end
```

Comment:
You can add a this method into `ApplicationController` for reusability purposes in different controllers